src/GitHub.App/Services/PullRequestService.cs

```diff
@@ -385,6 +413,23 @@ string CreateUniqueRemoteName(IRepository repo, string name)
             return uniqueName;
         }
 
+        async Task<string> ExtractToTempFile(IRepository repo, string commitSha, string fileName, Encoding encoding)
+        {
+            var contents = await gitClient.ExtractFile(repo, commitSha, fileName) ?? string.Empty;
+            return CreateTempFile(fileName, commitSha, contents, encoding);
+        }
+
+        static string CreateTempFile(string fileName, string commitSha, string contents, Encoding encoding)
+        {
+            var tempDir = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+            var tempFileName = $"{Path.GetFileNameWithoutExtension(fileName)}@{commitSha}{Path.GetExtension(fileName)}";
```

Comment:
At the moment I don't think the file path is surfaced anywhere, but it is visible in the #1036 PR (which isn't ideal).

![image](https://user-images.githubusercontent.com/11719160/27917573-42d479ec-6264-11e7-892d-c23b2abd503d.png)

Maybe we should find a way to sanitize the file name and not include the `commitSha`? Would URL-encoding the filename work? I think we definitely need to need to keep the file extension for automatic syntax highlighting.